test/units/TEST-07-PID1.working-directory.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+set -eux
+set -o pipefail
+
+# shellcheck source=test/units/util.sh
+. "$(dirname "$0")"/util.sh
+
+(! systemd-run --wait -p DynamicUser=yes \
+                      -p EnvironmentFile=-/usr/lib/systemd/systemd-asan-env \
+                      -p WorkingDirectory=~ true)
+
+assert_eq "$(systemd-run --pty --uid=nobody -p WorkingDirectory=~ pwd)" "/"
```

Comment:
Maybe, also better to add test cases for `--uid=testuser`, `--uid=root`, and dynamic user with static uid ?